### DISCUSSION TITLE
共享农场第5期优化需求计划（待确认）

### DISCUSSION THREAD

**Discussion by welllom** [famland/website#138] [Ideas]
## 5.1 基本田做市手续费调整。

取消基本田二次做市收取做市本金100%手续费规则。调整为：

只要famland 的 activeted=0  或 activeted=1，耕种时长≠0，基本田做市无手续费。

只要famland 的activeted=1且耕种时长= 0，（基本田做市后并撤市的famland）手续费按照当次做市本金U价值计收20%的手续费。

例如：
1、一个已经做市100U本金的基本田，再增加100U做市本金，无手续费;
2、一个已经基本田撤市的famland，再次做市200U价值的USDT/YL本金，将支付200*20%=20USDT+20USDT等额yl的手续费。

**需求优先级：1**

## 5.7 基本田撤市，做市本金赔付规则调整。

撤销基本田做市后，用户除了得到做市LP TOken对应的资产数额外，还获得的基本田赔付 = 赔付额度-做市本金价值 （获得的赔付最高为500U等额的YL，最低为0）

例如：一个满额做市的基本田，做市存入价值500U的LPtoken，因为YL价格下跌导致做市本金价值降低到400U。

如果用户已经主动或被动提取了累计价值100U的收益，赔付额度从500降低到400。此时，用户撤资基本田后，可以获得的是400U的LPtoken，还可以获得的赔付=400-400=0。

如果用户始终没有主动或被动提取过收益，赔付额度仍然为500，用户撤资基本田后可以获得400U的LPtoken，还可以获得的赔付=500-400=100U等额数量的YL。

**需求优先级：2**

## 5.8 基本田撤市手续费调整。

基本田撤市手续费由1%调整到5%。

**需求优先级：3**

## 5.3 农田进出农场收费规则调整。

取消农场主移除成员田的功能。改为5.9需求。

共享农场未做过广告（ad=0）或已做了广告（ad=1）且有效期已截止时，农田加入农场或农田主动从农场退出时，均要支付120YL。

共享农场ad=1且在广告有效期内，农田主动加入农场或农田主动从农场退出时，均要支付20YL。

以上只涉及手续费率调整，手续费的分流规则不变。

**需求优先级：4**

## 5.2  共享农场广告费收费规则调整。

取消农场广告到期后广告费，再次做广告翻倍的规则，不论是农场首次做广告，还是广告到期再次做广告，广告费率均按以下标准执行：
30天  900YL（日均30YL），90天  2500YL（日均27.78YL），180天  4500YL（日均25YL），360天  6000YL  （日均16.67YL）

**需求优先级：5**

## 5.4 新增撤销基本田做市对共享值影响的效能

基本田撤市时，将根据撤市本金数额扣减同等数量的共享值。

例如：某个Famland的基本田共享值为560，做市本金为100U的USDT/YL ，基本田撤市后，该Famland的共享值将为560-100=460。

如果玩家继续再这个Famland基本田新增做市200U的USDT/YL，那么Famland共享值仍然可以提升200，达到460+200=660。

**需求优先级：6**

## 5.5 新增Famland有效共享值对共享农场流速影响

主要是针对早期bug田的。基本田做市本金小于500U的USDT/YL，共享值又高于做市本金的，按实际做市本金计算有效共享值。只有做市本金大于等于500U的，共享值才按实际共享值额度计算流速。

例如：Famland基本田做市本金是100U的USDT/YL , 而其共享值为650，那么农场计算共享值时，不按650计算共享值，而是按照100计算有效共享值。如果基本田做市本金达到500U的USDT/YL，则农场计算总共享值时，才按照650计算有效共享值。

**需求优先级：7**

## 5.6 调整新Famland创建及收费机制。

新创建Famland须支付10YL。取消新建田其他条件限制。

**需求优先级：8**

## 5.9 新增农场主可限制所在农场成员Famland收益的功能。

农场主可定期更新最低Famland收益共享值指标，农场内共享值低于该共享值的Famland将显示在该列表中，并可手动根据需要对该Famland进行停止收益的操作。

农场主可以对其农场成员的Famland做出停止收益操作，被标注停止收益的Famland，其待收取收益的APY将被立即置0。只有农场主再次对该成员Famland做出恢复收益操作后，该Famalnd的APY才会恢复到农场当前APY计算收益。

**需求优先级：9**

说明：
1、以上方案优化的内容是社区自愿者，综合近期农场主和玩家反馈的意见，内部形成意见共识后做出的决议性提案。
2、公示后3天内，所有玩家均可以对公布的方案内容可以自由发表个人的意见。
3、公示期结束后，若未有争议的事项将开发部署，存在争议或无法暂确定的方案，将组织社区投票。具体投票规则可另行设定。

**Comment by welllom** [comment]
转志愿者说：

任何一个人或者国家不论取得多大的成就，有多少成熟的经验，也要容忍别的人或国家采取不同于自己的做法。人和国家不分大小、强弱、贫富，都应享有平等的权利，有充分表达自己意见的自由。涉及大事，应尽可能达成最大的共识，不同文明、不同的社会制度，应互相取长补短，在求同存异中发展。我们不仅要承认有共同利益，而且要努力去增大共同利益。这才是和合之道，也是中华文明能持续的核心价值。

我们这些自愿者，包括早期的创始人也好，前面经历了那么多次的争议，这么多次的问题探讨，能把更多人意见达成共识的东西，用区块链和智能合约的方式沉淀下来，就是在布道，砥砺前行。

**Comment by yancanlou** [comment]
5.4，
要进行分类，否则会大家都去撤田再做市而乱！
例：
手上有共享值大于等于495的激活过的撤市田。再次做市必须先减去495或500共享值，再加现做市金。
即只保留人工升值的部分的这个属性，资金的部分已随资金而去。
手上小于495的激活过的撤市田，应当于再次做市时减去原撤市金额，再加现做市金。

**Comment by yancanlou** [comment]
5.2广告费太高
按现有不加倍就是极限高，去了加倍，然后升额？
建议：
去除加倍，同时现有费额减半，变成1个月300，而不是1个月900，没好事谁愿意去打广告？
否则建农场的成本将变成比现在还高，这个是，苛政猛于虎！！！
广告费900+登记费120+600（新增入场费）-原生分发池贡献600=1620-600=1020成本
而现在算下来，
广告费600+登记费120-原生分发池贡献600=120成本！！！
建议改：
广告费0（首次激活赠送）+登记费120+600（新增入场费）-原生分发池贡献600=120。

**Comment by famland16816** [comment]
5.2
建议：
不收广告费。把这个，入场门槛交给农场主。
就是，像设置，这个共享值门槛一样，可以设置一个入场的费用。
这个费用是进入农场分发池的，大家共享的，按照农场收益规则进行再分配。就不用从原生农场搞600过来了。
相当于大家众筹把这个农场激活。因为新农场他分发池没有量是产生不了收益的。
至于这个费用收多少，农场主在组建农场时可以和，参与的农户，商讨决定。

**Comment by yancanlou** [comment]
5.4此项优先级最少与5.1同级，并同步优化！必对历史上已撤市田做市自动减共享值内不存在的原做市资本金加成，不然引起争撤田，生态错乱！！！

**Reply by otto-001** [reply]
现行规则撤市，虽然不扣共享值，但再次做市也不会再增加共享值。

**Comment by otto-001** [comment]
## 5.3 农田进出农场收费规则调整。

取消农场主移除成员田的功能。改为5.9需求。

共享农场未做过广告（ad=0）或已做了广告（ad=1）且有效期已截止时，农田加入农场或农田主动从农场退出时，均要支付30YL。

共享农场ad=1且在广告有效期内，农田主动加入农场或农田主动从农场退出时，均要支付3YL。

以上只涉及手续费率调整，手续费的分流规则不变。

**开发优先级：6**

## 5.2  共享农场广告费收费规则调整。

取消农场广告到期后广告费，再次做广告翻倍的规则，不论是农场首次做广告，还是广告到期再次做广告，广告费率均按以下标准执行：
30天  600YL（日均20YL），90天  1700YL（日均17.778YL），180天  2700YL（日均15YL），360天  4300YL  （日均12YL）

**开发优先级：7**

**Comment by otto-001** [comment]
5.6 调整新Famland创建及收费机制。
新创建Famland须支付10YL。取消新建田其他条件限制。
 
需求优先级：8

针对5.6需求建议如下：
优化合并创建Famland和做市的流程，在做市的时候，创建famland连同做市一气呵成。
即玩家存入做市资金的同时，就创建第一块famland,并完成基本田做市。
